packages/@aws-cdk/aws-kms/lib/key.ts

```diff
@@ -428,6 +542,8 @@ export class Key extends KeyBase {
       description: props.description,
       enableKeyRotation: props.enableKeyRotation,
       enabled: props.enabled,
+      keySpec: props.keySpec,
+      keyUsage: props.keyUsage,
```

Comment:
One thing to consider here: as-is, the implementation will mean that every customer with a key today will see an update for each of their keys when the upgrade the CDK, with both of these properties created. Given that these are the default values anyway, that _should_ be ok, just potentially mildly annoying.

The alternative -- which is more backwards-compatible -- would be to leave these as undefined if the customer does not specify them. I think I have a slight preference for that.

Reply:
I agree with you. So, I copied them to local variables for validation to keep the `props` unchanged, and existing tests passed without change. Should I implement in different way?

https://github.com/dacci/aws-cdk/blob/98231949897402007317c2b5ed89dce1514587e4/packages/%40aws-cdk/aws-kms/lib/key.ts#L507-L508

Reply:
Nope -- this is perfect. I must have mis-read it the first time. Sorry about that. 🤦‍♂️